src/Spago/Build.hs

```diff
@@ -87,10 +89,26 @@ build BuildOptions{..} maybePostBuild = do
 repl :: Spago m => [Purs.SourcePath] -> [Purs.ExtraArg] -> m ()
 repl sourcePaths passthroughArgs = do
   echoDebug "Running `spago repl`"
-  config <- Config.ensureConfig
-  deps <- Packages.getProjectDeps config
-  let globs = Packages.getGlobs deps <> Config.configSourcePaths config <> sourcePaths
-  Purs.repl globs passthroughArgs
+
+  hasPackagesDhall <- testfile "packages.dhall"
```

Comment:
I think here we could use the [`try`](https://www.stackage.org/haddock/lts-13.25/base-4.12.0.0/Control-Exception.html#v:try) pattern on the config to make this cleaner, something like:

```hs
try (Config.ensureConfig) >>= \case
  Right config -> do
    -- start the normal project repl with the `config`
    ...
  Left err -> do
    -- here we do the global repl stuff
    echoDebug err
    ...
```

The reason is that even if we get the boolean right in the `testfile` the `ensureConfig` might still except, but instead we can just catch the exception with the `try`

Reply:
Agree! Actually didn’t even consider the case where another exception might occur